src/providers/redux/userSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { IUserState } from '~/@types/redux';
+
+const initialState: IUserState = {
+  id: null,
+  name: '',
+  surname: '',
+  email: '',
+};
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    updateUser: (state, action: { payload: IUserState }) => {
```

Comment:
maybe we could add `UpdateUserAction` type and use it with `action: UpdateUserAction` ?

Reply:
I created interface:
```
export interface ActionPayload<T> {
  payload: T;
}
```
to be easily used with other reducers and changed line above to:
`action: ActionPayload<UserState>`